lib/std/os/windows.zig

```diff
@@ -1493,6 +1493,20 @@ pub fn VirtualFree(lpAddress: ?LPVOID, dwSize: usize, dwFreeType: DWORD) void {
     assert(kernel32.VirtualFree(lpAddress, dwSize, dwFreeType) != 0);
 }
 
+pub const VirtualProtectError = error{
+    InvalidAddress,
+    Unexpected,
+};
+
+pub fn VirtualProtect(lpAddress: ?LPVOID, dwSize: SIZE_T, flNewProtect: DWORD, lpflOldProtect: *DWORD) VirtualProtectError!void {
+    if (kernel32.VirtualProtect(lpAddress, dwSize, flNewProtect, lpflOldProtect) == 0) {
```

Comment:
Did you have a look to check if it's possible to reimplement `VirtualProtect` using Zig by calling directly to `ntdll`? In the first iteration, I am happy to go with `kernel32` however ideally we would avoid calling into the high level DLLs if possible.

Reply:
Ah, looks like it's possible with `NtProtectVirtualMemory` - I'll take a look at that in a bit

Reply:
Should I keep the `VirtualProtect` wrapper around for direct usage if anyone needs it, or should I remove it?

Reply:
If you're going to swap out the `kernel32` for `ntdll` import, having a wrapper that converts a Windows error code to Zig error is very valuable. Is that what you are asking for here?

Reply:
No, I was just asking if I should actually swap it out or if I should keep the `VirtualProtect` extern decl & wrapper around anyway - I assume the former since the signatures of the two are near-identical